src/components/NewBook.jsx

```diff
@@ -1,18 +1,31 @@
+import axios from 'axios';
 import { useState } from 'react';
 import { useDispatch } from 'react-redux';
 import { addBook } from '../redux/books/books';
+import { baseUrl, generateID } from '../redux/store';
+
+const initialState = {
+  item_id: generateID(),
+  title: '',
+  author: '',
+  category: 'Fiction',
+};
 
 export default function NewBook() {
-  const [formData, setFormData] = useState({
-    title: '',
-    category: 'Fiction',
-  });
+  const [formData, setFormData] = useState(initialState);
 
   const dispatch = useDispatch();
 
-  const handleSubmit = (e) => {
+  const handleSubmit = async (e) => {
     e.preventDefault();
-    dispatch(addBook(formData));
+    await axios.post(baseUrl, formData)
+      .then(() => {
+        dispatch(addBook(formData));
+        setFormData({
+          ...initialState,
+          item_id: generateID(),
+        });
```

Comment:
- [ ] Same here, Please use createAsyncThunk for creating thunk add book action creator, and kindly move it to the books redux slice and import and use it here in handleSubmit function.